### ISSUE TITLE
Clean up RxFire's API documentation 

### ISSUE THREAD

**Issue by shelcindra** [firebase/firebase-js-sdk#2572]
Firebase SDK version: rxfire

I'd like to clean up some of the rxfire docs.  Here are some things I would like to do:
*  Fix format error: Function parameters not displayed.  
* `collectionData` function name displayed incorrectly.
* Clean up examples: some are hard to follow, some have extra imports.
* Update descriptions to clarify function usage.

**Comment by rommelpe** [comment]
Thanks for the PR on the other request, @shelcindra. You can submit a PR for this one too, and we're excited to look into it.
